elastic/beats
# [AWS] Support owning account for cross account monitoring

## Proposed commit message

Adds `owning_account` configuration option for AWS where it accepts a source account ID. Internally, this maps to the ListMetrics API's [^1] `OwningAccount` parameter. Note that `include_linked_accounts` should be set to `true` to use the account filtering. 

```yaml
include_linked_accounts: true # default to true, hence can be ignored in configuration yaml
owning_account: 111111111111
```

See the screenshot below from an actual run with owning account filtering, 

![Screenshot 2024-09-09 at 10 26 11 AM](https://github.com/user-attachments/assets/11783db4-f583-4ddb-92c6-bd2c0c6bc0d7)

In comparison, consider the screenshot below where both monitoring account and linked accounts are visible as there is no filtering for `owning_account`, 

![Screenshot 2024-09-10 at 9 48 42 AM](https://github.com/user-attachments/assets/e2426205-279d-4f98-a1c9-49329b275271)


## Checklist


## Disruptive User Impact

None :`owning_account` is a new configuration 

## How to test this PR locally

- Build x-pack metricsbeat 
- Enable AWS monitoring with `include_linked_accounts` set to true (default is already true) & `owning_account` account ID
- Observe metrics data and validate that `cloud.account.id` match with `owning_account` ID

## Related issues

Closes #40570

[^1]:  https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_ListMetrics.html#API_ListMetrics_RequestParameters
